action/protocol/poll/staking_command.go

```diff
@@ -27,6 +27,7 @@ type stakingCommand struct {
 	stakingV1   Protocol
 	stakingV2   *staking.Protocol
 	candIndexer *CandidateIndexer
+	sr          protocol.StateReader
```

Comment:
I think it is weird to store sr in protocol struct

Reply:
agree, to have sr in CreatePostSystemActions, you can modify the API to 
`CreatePostSystemActions(ctx context.Context, sr StateReader) ([]action.Envelope, error)`

Reply:
address it